src/python/pants/build_graph/address.py

```diff
@@ -121,6 +122,10 @@ class Address:
     def parse(cls, spec: str, relative_to: str = "", subproject_roots=None) -> "Address":
         """Parses an address from its serialized form.
 
+        Note that this does not work properly with generated subtargets, e.g. the address
+        `helloworld/app.py`. We would not be able to calculate the `generated_base_target_name`, so
+        we treat this like a normal target address.
```

Comment:
We should maybe open a followup ticket with a short blurb describing the open question here (and referencing the doc for more info).

Reply:
I don't think this is necessary. I'm actively iterating on this part of the codebase the next day or two. It's possible that we adapt `Address.parse()` to be able to parse file names, given a hint for `generated_base_target_name`. But it's also possible that we want that logic to live somewhere else, like the dependency inference rule. Unclear, but will be answered very soon.